otherlibs/dynlink/Makefile

```diff
@@ -192,7 +192,7 @@ $(LOCAL_SRC)/%.cmx: $(LOCAL_SRC)/%.ml
 byte/dynlink_compilerlibs.cmo: $(COMPILERLIBS_CMO)
 	$(OCAMLC) $(COMPFLAGS) -pack -o $@ $(COMPILERLIBS_CMO)
 
-byte/dynlink_compilerlibs.cmi: byte/dynlink_compilerlibs.cmo
+byte/dynlink_compilerlibs.cmi: byte/dynlink_compilerlibs.cmo ;
```

Comment:
I would welcome a comment in the code to explain (some of) what is going on here.

Reply:
A good point - now that the PR is open, can I get away with referencing it, so something like:

```make
# The semi-colon creates an empty recipe for byte/dynlink_compilerlibs.cmi,
# which allows GNU make to understand that byte/dynlink_compilerlibs.cmi is
# generated as a side-effect of (re)building byte/dynlink_compilerlibs.cmo.
# If this is stated as a dependency only (i.e. without the semicolon) then
# parallel rebuilds can fail with inconsistent assumption errors (see #11180
# for a complete explanation).
```